### PR TITLE
Fixing Mint with External URI Event and Adding Corresponding Tests

### PR DESCRIPTION
**Pull request by asiniscalchi**
## PR Type:
Bug fix, Tests

___
## PR Description:
This PR addresses a bug in the minting process with an external URI event. It modifies the event logging mechanism and adds a new test case to ensure the correct behavior of the minting process. The changes are primarily in the `lib.rs` and `tests.rs` files of the `ownership-chain/precompile/laos-evolution` module.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`ownership-chain/precompile/laos-evolution/src/lib.rs`: The changes in this file mainly revolve around the minting process with an external URI. The logging mechanism for the minting event has been modified. Instead of using a `Log` struct, a `LogsBuilder` is used to record the event. This change provides a more flexible and comprehensive way to log events. Also, some unused imports have been removed.
`ownership-chain/precompile/laos-evolution/src/tests.rs`: A new test case `mint_with_external_uri_should_generate_log` has been added. This test ensures that the minting process with an external URI generates the correct log. The test uses a mock to simulate the minting process and checks the resulting log for correctness.
</details>

### CODE REVIEW

File: ownership-chain/precompile/laos-evolution/src/lib.rs [hunk 1]
```diff
@@ -124,18 +126,18 @@ where
 						let mut token_id_bytes = [0u8; 32];
 						token_id.to_big_endian(&mut token_id_bytes);
```
Comment (magecnion) [inline ownership-chain/precompile/laos-evolution/src/lib.rs:127]: `token_id_bytes` is needed anymore

**Review by magecnion — CHANGES_REQUESTED** [review]

File: ownership-chain/precompile/laos-evolution/src/lib.rs [hunk 2]
```diff
@@ -124,18 +126,18 @@ where
...
-						}
-						.record(handle)?;
+						LogsBuilder::new(context.address)
+							.log2(
+								SELECTOR_LOG_MINTED_WITH_EXTERNAL_TOKEN_URI,
+								to,
+								EvmDataWriter::new()
+									.write(collection_id)
+									.write(slot)
+									.write(Bytes(token_uri_raw))
```
[the remark this answers is not in the document]
Comment (asiniscalchi) [reply ownership-chain/precompile/laos-evolution/src/lib.rs:133]:
``` 
/// The `bytes`/`string` type of Solidity.
/// It is different from `Vec<u8>` which will be serialized with padding for each `u8` element
/// of the array, while `Bytes` is tightly packed.
#[derive(Clone, Debug, Eq, PartialEq)]
pub struct Bytes(pub Vec<u8>);
```

File: ownership-chain/precompile/laos-evolution/src/tests.rs [hunk 3]
```diff
@@ -108,6 +108,54 @@ fn create_collection_should_generate_log() {
...
+	let result = Mock::execute(&mut handle);
+	assert!(result.is_ok());
+	let logs = handle.logs;
+	assert_eq!(logs.len(), 1);
+	assert_eq!(logs[0].address, H160::zero());
+	assert_eq!(logs[0].topics.len(), 2);
+	assert_eq!(logs[0].topics[0], SELECTOR_LOG_MINTED_WITH_EXTERNAL_TOKEN_URI.into());
+	assert_eq!(
+		logs[0].topics[1],
+		H256::from_str("0x0000000000000000000000000101010101010101010101010101010101010101")
```
[the remark this answers is not in the document]
Comment (asiniscalchi) [reply ownership-chain/precompile/laos-evolution/src/tests.rs]: it's an H256 ... however I introduced ALICE

**Review by magecnion — APPROVED** [review]

**Pull request merged**